### ISSUE TITLE
getCarrier() must be of the type string or null error when creating shipment 

### ISSUE THREAD

**Issue by bojanni** [myparcelnl/magento#477]
**Describe the bug**
Error when creating consigment 

**To Reproduce**
1. Go to a placed order with status processing
2. Click on shipment
3. Scroll down to place the shipment

fatal error: Uncaught TypeError: Return value of MyParcelNL\Sdk\src\Adapter\DeliveryOptions\AbstractDeliveryOptionsAdapter::getCarrier() must be of the type string or null, boolean returned in /home/***/public_html/vendor/myparcelnl/sdk/src/Adapter/DeliveryOptions/AbstractDeliveryOptionsAdapter.php:88 Stack trace: #0 /home/***/public_html/vendor/myparcelnl/magento/Model/Sales/TrackTraceHolder.php(167): MyParcelNL\Sdk\src\Adapter\DeliveryOptions\AbstractDeliveryOptionsAdapter->getCarrier() #1 /home/***/public_html/vendor/myparcelnl/magento/Observer/NewShipment.php(105): MyParcelNL\Magento\Model\Sales\TrackTraceHolder->convertDataFromMagentoToApi(Object(Magento\Sales\Model\Order\Shipment\Track), Array) #2 /home/bvdhe1de/public_html/vendor/myparcelnl/magento/Observer/NewShipment.php(79): MyParcelNL\Magento\Observer\NewShipment->setMagentoAndMyParcelTrack(Object(Magento\Sales\Model\Order\Shipment)) #3 /home/****/public_html/vendor/magento/framework/Event/Invoker/InvokerDefault.php(88): MyParcelNL\Magento\Ob in /home/****/public_html/vendor/myparcelnl/sdk/src/Adapter/DeliveryOptions/AbstractDeliveryOptionsAdapter.php on line 88

**Expected behavior**
The shipment to be created sccesfully

**Screenshots**

**Desktop (please complete the following information):**
 - OS: CENTOS 7
 - Browser Opera / Edge / Chrome
 - MyParcel Plugin version 4.1.2
 - MyParcel SDK version 5.2.1

**Additional context**
It look like creating a shipment from the grid works. But more testing is required

looks like the same error from here https://github.com/myparcelnl/magento/issues/464#issuecomment-679156867

**Comment by CoenPonsen1984** [comment]
Did you manage to solve this issue? After updating to  MyParcel 4.1.2 we get this error. We can only create shipments on the order grid but no longer on the order page itself.

**Comment by bojanni** [comment]
Hi. 
No, we reverted back to 3.4 because of the various issues with 4.1.2

**Comment by RichardPerdaan** [comment]
Yesterday a new version of the MyParcel plugin was released. This version should solve this problem. 
This is why I will close the issue.

My advice is to use ven version 4.1.3

**Comment by CoenPonsen1984** [comment]
Shouldn't be closed issues persist

**Comment by gerben86** [comment]
Same issue overhere, Did you find a solution @CoenPonsen1984  or @bojanni ?

**Comment by bojanni** [comment]
@gerben86 we found that this error only occurs with orders that were placed with an older version the plugin. Orders placed with >v4.1.2 installed don't cause errors anymore in our store.

**Comment by gerben86** [comment]
> @gerben86 we found that this error occurs when we tried to create labels for orders that were placed with an older version the plugin installed. Orders placed with >v4.1.2 installed don't cause errors anymore in our store.

Yes, we encountered the samething. But it's sad that the plugin didn't take care of those, normal, situations.

**Comment by gjportegies** [comment]
One of our customers is experiencing this issue as well on MyParcel 4.1.4

Magento 2.3.5-p2
MyParcel 4.1.4
MyParcel SDK 5.2.4

```
2020/12/30 14:25:10 [error] 3041053#3041053: *12901626 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught TypeError: Return value of MyParcelNL\Sdk\src\Adapter\DeliveryOptions\AbstractDeliveryOptionsAdapter::getCarrier() must be of the type string or null, bool returned in /home/domains//vendor/myparcelnl/sdk/src/Adapter/DeliveryOptions/AbstractDeliveryOptionsAdapter.php:88
Stack trace:
#0 /home/domains//vendor/myparcelnl/magento/Model/Sales/TrackTraceHolder.php(171): MyParcelNL\Sdk\src\Adapter\DeliveryOptions\AbstractDeliveryOptionsAdapter->getCarrier()
#1 /home/domains//vendor/myparcelnl/magento/Observer/NewShipment.php(105): MyParcelNL\Magento\Model\Sales\TrackTraceHolder->convertDataFromMagentoToApi(Object(Magento\Sales\Model\Order\Shipment\Track), Array)
#2 /home/domains//vendor/myparcelnl/magento/Observer/NewShipment.php(79): MyParcelNL\Magento\Observer\NewShipment->setMagentoAndMyParcelTrack(Object(Magento\Sales\Model\Order\Shipment))
#3 /home/domains//vendor/magento/fram" while reading response header from upstream, client: 91.132.241.178, server: www., request: "POST //admin/order_shipment/save/order_id/114/key/6ff011839e887c28b5e565c764d63337342ecc04220cb11506e4845f569cf14a/ HTTP/1.1", upstream: "fastcgi://unix:/run/php-fpm/php-fpm.sock:", host: "www.", referrer: "/admin/order_shipment/new/order_id/114/key/<key>/"
```

**Comment by RichardPerdaan** [comment]
Today I managed to reproduce this problem. I am now working on the solution for this. 

The fix will be included in the next version of the MyParcel plugin (4.1.5-beta.2)

**Comment by gjportegies** [comment]
@RichardPerdaan any information on when the next version will be released?

**Comment by RichardPerdaan** [comment]
Solved in [4.1.6-beta.1](https://github.com/myparcelnl/magento/releases/tag/4.1.6-beta.1)
